kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -254,14 +280,21 @@ object Channel:
 
     /** WARNING: Low-level API meant for integrations, libraries, and performance-sensitive code. See AllowUnsafe for more details. */
     object Unsafe:
+        private enum Put[A]:
+            val promise: Promise.Unsafe[Closed, Unit]
+            case Batch(batch: Chunk[A], override val promise: Promise.Unsafe[Closed, Unit])
+            case Value(value: A, override val promise: Promise.Unsafe[Closed, Unit])
+        end Put
```

Comment:
This is very hot code so minimizing the memory footprint as much as possible would be ideal. It's possible to avoid an object header + a field pointer by extending from `IOPromise` directly:

```scala
sealed abstract class Put[A] extends IOPromise[Closed, Unit]
object Put:
   class Batch(batch: Chunk[A]) extends Put[A]
   class Value(value: A) extends Put[A]
```

We can do that as a follow up, though.

Reply:
wow wild!